plotAfb.py

```diff
@@ -0,0 +1,287 @@
+                                                        tempDenom.Sumw2()
+                                                        tempNom.Sumw2()
+                                                        tempNom=tempNom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+                                                        tempDenom=tempDenom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+                                                        tempHistAfb=ROOT.TH1D("Afb"+flavor+year+category+dataType,"The distribution of Afb for "+year+" "+flavor+" result",len(binning(flavor))-1  ,array("f",binning(flavor)))
+                                                        #print(tempNom.GetNbinsX())
+                                                        #print(tempDenom.GetNbinsX())
+                                                        tempHistAfb.Divide(tempNom,tempDenom)
+                                                        histTree[flavor][year][dataType][category]=tempHistAfb.Clone()
+						else:
+							tempHistPos=TheStack(backgroundTree[flavor][year],lumi[flavor][year],plotTree[flavor][category][0],zScaleFac[flavor][year]).theHistogram
+							tempHistNeg=TheStack(backgroundTree[flavor][year],lumi[flavor][year],plotTree[flavor][category][1],zScaleFac[flavor][year]).theHistogram
+							tempNom=tempHistPos.Clone()
+							tempDenom=tempHistPos.Clone()
+							tempNom.Add(tempHistNeg,-1)
+							tempDenom.Add(tempHistNeg)
+							tempDenom.Sumw2()
+							tempNom.Sumw2()
+							tempNom=tempNom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+							tempDenom=tempDenom.Rebin(len(bng) - 1, 'hist_' + uuid.uuid4().hex, array('d', bng))
+							tempHistAfb=ROOT.TH1D("Afb"+flavor+year+category+dataType,"The distribution of Afb for "+year+" "+flavor+" result",len(binning(flavor))-1  ,array("f",binning(flavor)))
+							#print(tempNom.GetNbinsX())
+                                                        #print(tempDenom.GetNbinsX())
+							tempHistAfb.Divide(tempNom,tempDenom)
+							histTree[flavor][year][dataType][category]=tempHistAfb.Clone()			
+	for flavor in ["muon","electron"]:
+		for year in ["2016","2017","2018"]:
+			for category in ["BB","BE","All"]:
+				hCanvas = TCanvas("hCanvas", "Distribution", 500,500)
+				plotPad = ROOT.TPad("plotPad","plotPad",0,0,1,1)
+				setTDRStyle()	
+				plotPad.UseCurrentStyle()
+				plotPad.Draw()
+				plotPad.cd()
+				#plotPad.SetLogx()
+				xMax=histTree[flavor][year]["mc"][category].GetXaxis().GetXmax()
+				xMin=histTree[flavor][year]["mc"][category].GetXaxis().GetXmin()
+				yMax=histTree[flavor][year]["mc"][category].GetBinContent(histTree[flavor][year]["mc"][category].GetMaximumBin())
+				yMin=0.01
+				legend=ROOT.TLegend(0.1,0.7,0.4,0.98)
+				histTree[flavor][year]["data"][category].SetMarkerStyle(3)
+				histTree[flavor][year]["mc"][category].SetFillColorAlpha(2,0.5)
```

Comment:
Please don't use a filled histogram here, just a single line. AFB is just a single number and the filled histogram does not really emphasizes that enough I think.